src/resolve/resolved-dns-transaction.c

```diff
@@ -1199,8 +1201,46 @@ void dns_transaction_process_reply(DnsTransaction *t, DnsPacket *p, bool encrypt
         case DNS_PROTOCOL_DNS:
                 assert(t->server);
 
+                r = dns_packet_ede_rcode(p, &ede_msg);
+                if (r < 0 && r != -EINVAL)
+                        goto fail;
```

Comment:
i probably would never fail here, but only write a debug log message:

```c
ede_rcode = dns_packet_ede_rcode(p, &ede_msg);
if (ede_rcode < 0 && ede_rcode != -EINVAL)
        log_debug_errno(ede_rcode, "Unable to extract EDE error code from packet, ignoring: %m");
```

and then proceed